torch/csrc/distributed/c10d/NCCLUtils.cpp

```diff
@@ -118,6 +118,14 @@ std::shared_ptr<NCCLComm> NCCLComm::create_scalable(
 #endif // NCCL_HAS_INIT_RANK_SCALABLE
 #endif // NCCL_HAS_CONFIG
 
+#ifdef NCCL_SIM_INFO_INITIALIZER
+float NCCLComm::ncclCollectiveEstimateEnd(
+    std::unique_ptr<ncclSimInfo_t> simInfo) {
+  C10D_NCCL_CHECK(ncclGroupSimulateEnd(simInfo.get()), std::nullopt);
+  return simInfo->estimatedTime;
+}
+#endif // NCCL_SIM_INFO_INITIALIZER
+
```

Comment:
```
float NCCLComm::ncclCollectiveEstimateEnd() {
  ncclSimInfo_t simInfo = NCCL_SIM_INFO_INITIALIZER;
  C10D_NCCL_CHECK(ncclGroupSimulateEnd(&simInfo), std::nullopt);
  return simInfo->estimatedTime;
}
```
so that this API's signature does not depend on `ncclSimInfo_t`

Reply:
Actually, since the implementation here does not use the comm, you can put `ncclGroupSimulateEnd` directly in `ProcessGroupNCCL` instead of `NCCLComm`.